### ISSUE TITLE
Mode "replace" does not delete "mods" field from context

### ISSUE THREAD

**Issue by 4ok** [bem/bem-xjst#269]
```js
block('blockName').mod('modName', 'modVal')(

    replace()((ctx, json) => {
        console.log(ctx.mods.modName);

        return {
            block: 'blockName',
            content: 'Content...',
        }
    })
);
```

Output:

`RangeError: Maximum call stack size exceeded`

Because — `ctx.mods.modName === 'modVal'`

**Comment by miripiruni** [comment]
> Mode "replace" does not delete "mods" field from context

Nope. I can’t reproduce it. Example: http://goo.gl/LExHnM

**Comment by miripiruni** [comment]
Which version of bem-xjst?

**Comment by miripiruni** [comment]
Oh. I see. v6.4.1.

Looks like it’s relevant to #262

**Comment by miripiruni** [comment]
It looks like wontfix because your replacer is the same block.

**Comment by 4ok** [comment]
>It looks like wontfix because your replacer is the same block.

Previously, it worked with the deletion. Your example http://goo.gl/LExHnM

**Comment by miripiruni** [comment]
@4ok because of fix #262 the behaviour was changed.

**Comment by 4ok** [comment]
@miripiruni I think what mode wrap and replace must has different behaviour.  IMHO mode `replace` needs to replace the context.

How can I call the block without modifier, from same block with modifier otherwise?

**Comment by zxqfox** [comment]
@miripiruni Perhaps, replace should not take mods for blocks from parents as well as elemMods for elems, no?

**Comment by miripiruni** [comment]
Fixed in v6.4.2
